### ISSUE TITLE
TypeError: type() argument 1 must be string, not newstr

### ISSUE THREAD

**Issue by clokep** [jaraco/configparser#30]
Thanks for making this package! Due to #27 I just upgraded configparser and 3.7.0, unfortunately it breaks flake8 on Python 2.7.15:

```
Traceback (most recent call last):
  File "/Users/pcloke/.virtualenvs/my_venv/bin/flake8", line 7, in <module>
    from flake8.main.cli import main
  File "/Users/pcloke/.virtualenvs/my_venv/lib/python2.7/site-packages/flake8/main/cli.py", line 2, in <module>
    from flake8.main import application
  File "/Users/pcloke/.virtualenvs/my_venv/lib/python2.7/site-packages/flake8/main/application.py", line 14, in <module>
    from flake8.main import options
  File "/Users/pcloke/.virtualenvs/my_venv/lib/python2.7/site-packages/flake8/main/options.py", line 4, in <module>
    from flake8.main import vcs
  File "/Users/pcloke/.virtualenvs/my_venv/lib/python2.7/site-packages/flake8/main/vcs.py", line 4, in <module>
    from flake8.main import mercurial
  File "/Users/pcloke/.virtualenvs/my_venv/lib/python2.7/site-packages/flake8/main/mercurial.py", line 7, in <module>
    import configparser
  File "/Users/pcloke/.virtualenvs/my_venv/lib/python2.7/site-packages/configparser.py", line 12, in <module>
    from backports.configparser import (
  File "/Users/pcloke/.virtualenvs/my_venv/lib/python2.7/site-packages/backports/configparser/__init__.py", line 139, in <module>
    from backports.configparser.helpers import OrderedDict as _default_dict
  File "/Users/pcloke/.virtualenvs/my_venv/lib/python2.7/site-packages/backports/configparser/helpers.py", line 193, in <module>
    dict(__metaclass__=abc.ABCMeta),
  File "/Users/pcloke/.virtualenvs/my_venv/bin/../lib/python2.7/abc.py", line 87, in __new__
    cls = super(ABCMeta, mcls).__new__(mcls, name, bases, namespace)
TypeError: type() argument 1 must be string, not newstr
```

Let me know if I can provide any other information! This seems to break on both Linux and macOS. Rolling back to 3.5.3 fixed this issue for me.

**Comment by demartinofra** [comment]
We are facing the same issue in https://github.com/aws/aws-parallelcluster.

We are importing configparser here: https://github.com/aws/aws-parallelcluster/blob/develop/cli/pcluster/easyconfig.py#L26

Steps to reproduce the issue:

- Create a virtualenv with python 2.7
- pip install aws-parallelcluster
- pcluster --help

Error:

```
pcluster --help
Traceback (most recent call last):
  File "/Users/fdm/.pyenv/versions/parallelcluster/bin/pcluster", line 6, in <module>
    from pcluster.cli import main
  File "/Users/fdm/.pyenv/versions/2.7.9/envs/parallelcluster/lib/python2.7/site-packages/pcluster/cli.py", line 22, in <module>
    from pcluster import easyconfig, pcluster
  File "/Users/fdm/.pyenv/versions/2.7.9/envs/parallelcluster/lib/python2.7/site-packages/pcluster/easyconfig.py", line 26, in <module>
    import configparser
  File "/Users/fdm/.pyenv/versions/2.7.9/envs/parallelcluster/lib/python2.7/site-packages/configparser.py", line 12, in <module>
    from backports.configparser import (
  File "/Users/fdm/.pyenv/versions/2.7.9/envs/parallelcluster/lib/python2.7/site-packages/backports/configparser/__init__.py", line 139, in <module>
    from backports.configparser.helpers import OrderedDict as _default_dict
  File "/Users/fdm/.pyenv/versions/2.7.9/envs/parallelcluster/lib/python2.7/site-packages/backports/configparser/helpers.py", line 193, in <module>
    dict(__metaclass__=abc.ABCMeta),
  File "/Users/fdm/.pyenv/versions/2.7.9/envs/parallelcluster/lib/python2.7/abc.py", line 87, in __new__
    cls = super(ABCMeta, mcls).__new__(mcls, name, bases, namespace)
TypeError: type() argument 1 must be string, not newstr
```

**Comment by nimbol** [comment]
FYI from what I can tell this stems from an incompatibility with `future` (0.17.1).

**Comment by gillesdouaire** [comment]
Same here, several of our builds are failing right now with that same error.

**Comment by naggie** [comment]
We have the same problem which causes flake8 to fail (even if the flake8 version is fixed)

**Comment by jaraco** [comment]
Good insight @nimbol.

With the latest commit on master, I'm able to replicate the issue in the test suite:

```
configparser master $ tox -e py27-future                                                                                                                                                 
GLOB sdist-make: /Users/jaraco/code/public/configparser/setup.py
py27-future inst-nodeps: /Users/jaraco/code/public/configparser/.tox/.tmp/package/1/configparser-3.7.0.zip
py27-future installed: DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7.,configparser==3.7.0,future==0.17.1,pathlib2==2.3.3,scandir==1.9.0,six==1.12.0
py27-future run-test-pre: PYTHONHASHSEED='4272121818'
py27-future runtests: commands[0] | /Users/jaraco/code/public/configparser/.tox/py27-future/bin/python test_configparser.py
Traceback (most recent call last):
  File "test_configparser.py", line 29, in <module>
    from backports import configparser
  File "/Users/jaraco/code/public/configparser/src/backports/configparser/__init__.py", line 139, in <module>
    from backports.configparser.helpers import OrderedDict as _default_dict
  File "/Users/jaraco/code/public/configparser/src/backports/configparser/helpers.py", line 193, in <module>
    dict(__metaclass__=abc.ABCMeta),
  File "/Users/jaraco/code/public/configparser/.tox/py27-future/bin/../lib/python2.7/abc.py", line 87, in __new__
    cls = super(ABCMeta, mcls).__new__(mcls, name, bases, namespace)
TypeError: type() argument 1 must be string, not newstr
ERROR: InvocationError for command '/Users/jaraco/code/public/configparser/.tox/py27-future/bin/python test_configparser.py' (exited with code 1)
________________________________________________________________________________________ summary ________________________________________________________________________________________
ERROR:   py27-future: commands failed
```

**Comment by jaraco** [comment]
Fix released as 3.7.1. Thanks for the report.

**Comment by clokep** [comment]
Version 3.7.1 worked, thanks for fixing this so fast! 👍
